contracts/pods/PodAdminGateway.sol

```diff
@@ -0,0 +1,179 @@
+        uint256 _podId,
+        AdminPriviledge priviledge,
+        bytes32 _tribeRole
+    )
+        public
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.ROLE_ADMIN,
+            TribeRoles.GUARDIAN
+        )
+    {
+        podAdminRoles[_podId][priviledge].remove(_tribeRole);
+        emit RevokePodAdminPriviledge(_podId, _tribeRole);
+    }
+
+    /// @notice Batch grant admin priviledges
+    function batchGrantAdminPriviledge(
+        uint256[] memory _podId,
+        AdminPriviledge[] memory priviledge,
+        bytes32[] memory _tribeRole
+    ) external hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.ROLE_ADMIN) {
+        for (uint256 i = 0; i < _podId.length; i++) {
+            grantPodAdminPriviledge(_podId[i], priviledge[i], _tribeRole[i]);
+        }
+    }
+
+    /// @notice Batch grant admin priviledges
+    function batchRevokeAdminPriviledge(
+        uint256[] memory _podId,
+        AdminPriviledge[] memory priviledge,
+        bytes32[] memory _tribeRole
+    )
+        external
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.ROLE_ADMIN,
+            TribeRoles.GUARDIAN
+        )
+    {
+        for (uint256 i = 0; i < _podId.length; i++) {
+            revokePodAdminPriviledge(_podId[i], priviledge[i], _tribeRole[i]);
+        }
+    }
+
+    ///// Expose admin functionality to addresses which have the appropriate TribeRole
+
+    /// @notice Admin functionality to add a member to a pod
+    function addMemberToPod(uint256 _podId, address _member) public {
+        require(_member != address(0), "ZERO_ADDRESS");
+        validateAdminPriviledge(_podId, AdminPriviledge.ADD_MEMBER, msg.sender);
+
+        memberToken.mint(_member, _podId, bytes(""));
+    }
+
+    /// @notice Admin functionality to batch add a member to a pod
+    function batchAddMemberToPod(uint256 _podId, address[] memory _members)
+        external
+    {
+        for (uint256 i = 0; i < _members.length; i++) {
+            addMemberToPod(_podId, _members[i]);
+        }
+    }
+
+    /// @notice Admin functionality to remove a member from a pod
+    function removeMemberFromPod(uint256 _podId, address _member) public {
+        require(_member != address(0), "ZERO_ADDRESS");
+
+        validateAdminPriviledge(
+            _podId,
+            AdminPriviledge.REMOVE_MEMBER,
+            msg.sender
+        );
+        memberToken.burn(_member, _podId);
+    }
+
+    /// @notice Admin functionality to batch remove a member from a pod
+    function batchRemoveMemberFromPod(uint256 _podId, address[] memory _members)
+        external
+    {
+        for (uint256 i = 0; i < _members.length; i++) {
+            removeMemberFromPod(_podId, _members[i]);
+        }
+    }
+
+    /// @notice Valdidate that a calling address has the relevant admin priviledge, for the
+    ///         function it is calling
+    function validateAdminPriviledge(
```

Comment:
I feel strongly that we should remove this function as it is pretty messy.

Reply:
What would you suggest replacing it with?

It does the job of checking whether a relevant `caller` has the relevant TribeRole which has the relevant admin priviledge for whatever the calling function requires. 

The task is a bit fiddly as we need to first find out what roles the calling address has, and then check if any of those roles has a relevant admin priviledge